e2e/cypress/support/task_commands.js

```diff
@@ -28,7 +28,12 @@ Cypress.Commands.add('postMessageAs', ({sender, message, channelId, rootId, crea
 * @param {Object} data - payload on incoming webhook
 */
 Cypress.Commands.add('postIncomingWebhook', ({url, data}) => {
-    cy.task('postIncomingWebhook', {url, data}).its('status').should('be.equal', 200).wait(TIMEOUTS.MEDIUM);
+    cy.task('postIncomingWebhook', {url, data}).its('status').should('be.equal', 200);
+
+    cy.waitUntil(() => cy.getLastPost().then((el) => {
```

Comment:
What if you have 2 or more tests that already posted and the intended incoming message hasn't come in yet? Wouldn't this get whatever the latest even if the current test hasn't posted yet?

Reply:
No, the retry starts with getting the last post then evaluating whether it has the expected value ``() => cy.getLastPost().then()``. So if the incoming message hasn't posted yet, it will retry to get the last post until it's true or timed out.

Reply:
If response of postIncomingWebhook is synchronous then my previous question is invalid, but if it is async, then I think it's valid.

Reply:
Ah I see .includes(data.attachments[0].pretext)). Cool